Core/Core/Configuration/Combine/Config/Config.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  Config.swift
+//  Core
+//
+//  Created by Muhammad Umer on 11/11/2023.
+//
+
+import Foundation
+
+public protocol ConfigProtocol {
+    var baseURL: URL { get }
+    var oAuthClientId: String { get }
+    var tokenType: TokenType { get }
+    var feedbackEmail: String { get }
+    var appStoreLink: String { get }
+    var agreement: AgreementConfig { get }
+    var firebase: FirebaseConfig { get }
+    var features: FeaturesConfig { get }
+}
+
+public enum TokenType: String {
+    case jwt = "JWT"
+    case bearer = "BEARER"
+}
+
+private enum ConfigKeys: String {
+    case baseURL = "API_HOST_URL"
+    case oAuthClientID = "OAUTH_CLIENT_ID"
+    case tokenType = "TOKEN_TYPE"
+    case feedbackEmailAddress = "FEEDBACK_EMAIL_ADDRESS"
+    case environmentDisplayName = "ENVIRONMENT_DISPLAY_NAME"
+    case platformName = "PLATFORM_NAME"
+    case organizationCode = "ORGANIZATION_CODE"
+    case appstoreID = "APP_STORE_ID"
+}
+
+public class Config {
+    public static let shared = Config()
+    
+    internal var properties: [String: Any] = [:]
+    
+    internal init(properties: [String: Any]) {
+        self.properties = properties
+    }
+    
+    private convenience init() {
+        self.init(properties: [:])
+        loadAndParseConfig()
+    }
+    
+    private func loadAndParseConfig() {
+        guard let path = Bundle.main.path(forResource: "config", ofType: "plist"),
+              let data = try? Data(contentsOf: URL(fileURLWithPath: path)),
+              let dict = try? PropertyListSerialization.propertyList(from: data, options: [], format: nil) as? [String: Any] else { return }
```

Comment:
Please fix the Swiftlint warning.
`Line Length Violation: Line should be 120 characters or less; currently it has 140 characters (line_length)`

-> 
```
guard let path = Bundle.main.path(forResource: "config", ofType: "plist"),
              let data = try? Data(contentsOf: URL(fileURLWithPath: path)),
              let dict = try? PropertyListSerialization.propertyList(
                from: data,
                options: [],
                format: nil) as? [String: Any] else { return }
```
or
```
guard let path = Bundle.main.path(forResource: "config", ofType: "plist"),
              let data = try? Data(contentsOf: URL(fileURLWithPath: path)),
              let dict = try? PropertyListSerialization.propertyList(
                from: data,
                options: [],
                format: nil
              ) as? [String: Any] else {
            return
        }
```